library/nametab.ml

```diff
@@ -228,12 +227,9 @@ let push visibility uname o tab =
 
 let rec remove_path uname tree = function
   | modid :: path ->
+      let modify _ mc = remove_path uname mc path in
       let map =
-        try
-          let submap = ModIdmap.find modid tree.map in
-          let submap = remove_path uname submap path in
-          if is_empty_tree submap then ModIdmap.empty
-          else ModIdmap.add modid submap tree.map
+        try ModIdmap.modify modid modify tree.map
```

Comment:
Shouldn't we still remove the subtree pointed by modid when it's empty? I don't know if we rely somewhere on the invariant that tries should not contain empty components.

Reply:
Why not. How would you do it? Using a specific version of modify taking an extra argument to check if the data has become irrelevant and contract  a `MNode {l=MEmpty; v; d; r=MEmpty; h}` for irrelevant `d` into a `MEmpty`?

Or to come back to the previous version, computing the map with the `modid` branch removed,  checking if both this new `map` and the new `submap` are empty, returning accordingly either `ModIdmap.empty` or the addition of the new `submap` to the modid-omitting `map`?

Reply:
Good idea, done.

Reply:
I think that going back to the previous code and removing the binding instead of returning the empty map is the simplest way to fix it.